scalalib/src/mill/javalib/android/AndroidAppModule.scala

```diff
@@ -47,13 +49,32 @@ object AndroidLintReportFormat extends Enumeration {
 @mill.api.experimental
 trait AndroidAppModule extends JavaModule {
 
+  final def root: os.Path = super.millSourcePath
+  private def src: os.Path = root / "src"
+  override def millSourcePath: os.Path = src / "main"
+
+  override def sources: T[Seq[PathRef]] = Task.Sources(millSourcePath / "java")
```

Comment:
you probably need to override this to `millSourcePath / "kotlin"` in the `AndroidAppKotlinModule`

Reply:
This took me by surprise as well, but this is how android studio + gradle structure the project

![Screenshot From 2024-12-28 13-53-38](https://github.com/user-attachments/assets/49ed99b6-a16e-47ce-8fd5-1f4659c353a4)

Reply:
This is rather a historical reason, because originally the primary language was Java, then, once Kotlin support was added, Kotlin files had to be under `main/kotlin` and then AGP added support of the mixed Java + Kotlin sources under the `main/java`.

If you try to create a new directory, `main/kotlin` will be in the dropdown list next to the `main/java` as well.

Anyway, since return type here is `T[Seq[PathRef]]`, it is definitely worth to have `Task.Sources(millSourcePath / "java")` + `Task.Sources(millSourcePath / "kotlin")` override in the Kotlin-specific module to support better ootb experience.

Reply:
makes sense, and I guess we could mix an android scala module in the future and make it work similarly. Let me make a change in the example too and test it out (with the Kotlin dir)

Reply:
To be precise, going until `src/main` is already enough, because unlikely it will be anything else with compilation sources under it other than `java` or `kotlin` (but AGP allows to defined custom source sets though) and `KotlinModule` will walk child hierarchy anyway https://github.com/com-lihaoyi/mill/blob/d663c0302f9474c4be924353462d7cc4d725f8b0/kotlinlib/src/mill/kotlinlib/KotlinModule.scala#L26-L44, but let's for now a strict whitelist of `java` and `kotlin` subfolders only.

Reply:
I've added the kotlin path and enriched the example, let me know if you'd like anything else, otherwise please resolve conversation

Reply:
LGTM, but I don't have an option to resolve this conversation.